src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -40,8 +46,15 @@
  *
  * @param <X> Type of item
  * @since 0.12
+ * @checkstyle ClassDataAbstractionCouplingCheck (550 lines)
  */
-public final class IterableOf<X> extends IterableEnvelope<X> {
+@SuppressWarnings("PMD.OnlyOneConstructorShouldDoInitialization")
+public final class IterableOf<X> implements Iterable<X> {
```

Comment:
@victornoel what's the reason for implementing `Iterable` now instead of keep extending `IterableEnvelope`?

Reply:
@scristalli it was kind of explained in the description: the envelope does not contain any specific behaviour that is implemented by `IterableOf`. `IterableOf` is not a decorator on `Iterable`, it's an implementation of `Iterable`. I could have extended `IterableEnvelope`, but I believe it would have been a design mistake semantically. Also, now that I think about it, it wouldn't have been possible to implement `toString`, `equals` and `hashCode` if I was extending `IterableEnvelope`.